Amelie/filter subjects

AmelieRolland opened the pull request:
- Affichage des matière selon le niveau selectionné
- niveau selectionné par onMouseHover
- ajout de style

![image](https://github.com/user-attachments/assets/6a13c93d-0353-4e07-b67a-81d425fb6413)

AmelieRolland committed:
création navbar

AmelieRolland committed:
vérification books !null

AmelieRolland committed:
modification navbar origine

AmelieRolland committed:
Merge remote-tracking branch 'origin/main' into amelie/navbar-home

AmelieRolland committed:
création navbar

 réalisation de la navbar qui accueillera un tri par niveau et par matière

AmelieRolland committed:
modifications

modifications gestion erreurs, et remplacement d'image si elle n'existe pas

AmelieRolland committed:
modifications

object-contain pour que l'image soit entière

AmelieRolland committed:
ajout schooltypes dans query books

AmelieRolland committed:
fetch schooltypes

AmelieRolland committed:
affichage des schooltypes et fix du loading

AmelieRolland committed:
affichage des matières

AmelieRolland committed:
sorting des matières par niveau

AmelieRolland committed:
tri des matières qui ont des livres

AmelieRolland committed:
ajout de css

asurare commented on src/components/Booklist.js:
```diff
@@ -1,45 +1,35 @@
-    const [selectedLevel, setSelectedLevel] = useState();
 
-    const { data: books } = useQuery({
+    const genericImageUrl ='/img/book.png';
```
Pas nécessaire que cette variable soit dans le composant

asurare commented on src/components/Booklist.js:
```diff
@@ -1,45 +1,35 @@
         return (
             <p>En attente du chargement</p>
         );
+    } else if (!books || books.length === 0 ){
```
Tu peux passer par ça 😉 
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Operators/Optional_chaining

AmelieRolland replied:
else if (books?.length === 0 )

asurare commented on src/components/Booklist.js:
```diff
@@ -1,45 +1,35 @@
+                <h1 className='permanent-marker text-3xl py-16'>Nos ouvrages  :</h1>
                 <div className="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-6">
-                    {filteredBooks.map((book) => (
+                    {books.map((book) => (
```
En temps normal, on va vérifier si quelque chose est disponible avant de l'afficher, or ici ce n'est effectivement pas nécessaire
Saurai-tu expliquer pourquoi ? 🤔

AmelieRolland replied:
Est ce que c'est parce que j'ai déjà anticipé le cas où books est inexistant?
![image](https://github.com/user-attachments/assets/8921c557-a2d1-4b00-a8e6-4534ff91d5d8)

asurare commented on src/components/Booklist.js:
```diff
@@ -1,45 +1,35 @@
                         <div key={book.id} className={"group flex flex-row transition duration-700 ease-in-out max-w-full h-auto border border-gray-200 rounded-lg shadow-lg dark:bg-gray-800 dark:border-gray-700" + (book.valid === true ? ' bg-white' : ' hidden')}>
                             <div className="flex-shrink-0 w-1/3">
-                                <img src={book.urlLite} alt={book.displayTitle} className="w-full h-full object-cover rounded-l-lg transform transition-transform duration-700 ease-in-out group-hover:rotate-6" />
+                                <img src={book.urlLite ? book.urlLite : genericImageUrl} alt={book.displayTitle} className="w-full h-full object-contain rounded-l-lg transform transition-transform duration-700 ease-in-out group-hover:rotate-6" />
```
"??" 😉

AmelieRolland replied:
src={book.urlLite ?? genericImageUrl}

asurare commented on src/components/SchoolTypesList.js:
```diff
@@ -0,0 +1,36 @@
+    });
+    if (isLoading) {
+        return <p>En attente de chargement</p>;
+    } else if (!dataSchool || dataSchool.length === 0) {
```
Idem ici 🙂

asurare commented on src/components/SubjectsList.js:
```diff
@@ -0,0 +1,55 @@
+        (subject.schoolTypes.find((schoolType) => schoolType.includes(selectedSchoolType)) && (subject.hasBooks === true)
+    )) : subjects;
+
+    // const filteredSubjects = selectSubjects.filter(selectSubject => selectSubject.hasBooks === true);
```
Si plus utilisé, tu peux supprimer les commentaires de code comme celui-ci

asurare commented on src/components/SubjectsList.js:
```diff
@@ -0,0 +1,55 @@
+    const handleSelectSchoolType = (schoolType) =>
+        setSelectedSchoolType(schoolType);
+
+    console.log(selectedSchoolType);
```
Console.log de debug oublié ? 🤔

asurare commented on src/data/api.js:
```diff
@@ -132,3 +145,12 @@ export const allSchoolTypes = async () => {
+export const allSubjects = async () => {
+  try {
+    const dataSubjects = await request(endpoint, SUBJECTS);
+    return dataSubjects.viewer.subjects.hits;
```
Es-tu certaine d'avoir cette suite avec tous les éléments définis quoi qu'il arrive ?

asurare commented on src/data/api.js:
```diff
@@ -132,3 +145,12 @@ export const allSchoolTypes = async () => {
+    const dataSubjects = await request(endpoint, SUBJECTS);
+    return dataSubjects.viewer.subjects.hits;
+  } catch (e) {
+    throw new Error("Erreur lors du chargement des écoles" + e);
```
Privilégié les template strings dans ce genre de cas :
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

AmelieRolland replied:
throw new Error(`Erreur lors du chargement des écoles" ${e}`);

AmelieRolland committed:
modifications

string template privilégié, coalescing operator, suppression de consolelog et commentaires obsoletes

AmelieRolland committed:
const img generique déplacée dans app.js
